lib/RawSource.js

```diff
@@ -51,17 +68,21 @@ class RawSource extends Source {
 	 * @returns {void}
 	 */
 	streamChunks(options, onChunk, onSource, onName) {
-		if (this._value === undefined) {
+		if (this._value === undefined && isDualStringBufferCachingEnabled()) {
 			this._value = Buffer.from(this._valueAsBuffer, "utf-8");
 		}
-		if (this._valueAsString === undefined) {
-			this._valueAsString =
+		let strValue = this._valueAsString;
+		if (strValue === undefined) {
+			strValue =
 				typeof this._value === "string"
 					? this._value
-					: this._value.toString("utf-8");
+					: internString(this._value.toString("utf-8"));
+			if (isDualStringBufferCachingEnabled()) {
+				this._valueAsString = strValue;
+			}
 		}
```

Comment:
This looks like we want to call `const strValue = this.source()`?

Reply:
The logic is a bit strange since `source()` can return a Buffer (when `convertToString = false`) but it looks a bit cleaner now.